server/src/main/java/org/elasticsearch/script/IpFieldScript.java

```diff
@@ -66,6 +67,13 @@ public final void runForDoc(int docId) {
         execute();
     }
 
+    public final void runForDoc(int docId, Consumer<InetAddress> consumer) {
+        runForDoc(docId);
+        for (int i = 0; i < count; i++) {
+            consumer.accept(InetAddressPoint.decode(values[i].bytes));
```

Comment:
Annoyingly, emit takes an InetAddress and encodes it into a BytesRef and this then immediately decodes it back to an InetAddress to pass to the indexer.  Possibly as a follow up it might be worth changing the script to store its values as InetAddresses and to decode them for doc values access only?

Reply:
makes sense to me